provisioners/src/main/java/org/jboss/intersmash/provision/helm/wildfly/eap81/Eap81HelmChartReleaseAdapter.java

```diff
@@ -0,0 +1,746 @@
+import org.jboss.intersmash.model.helm.charts.values.eap81.S2i;
+import org.jboss.intersmash.model.helm.charts.values.eap81.Tls;
+import org.jboss.intersmash.model.helm.charts.values.eap81.Tls__1;
+import org.jboss.intersmash.provision.helm.HelmChartReleaseAdapter;
+import org.jboss.intersmash.provision.helm.Image;
+import org.jboss.intersmash.provision.helm.wildfly.WildflyHelmChartRelease;
+
+import com.google.common.base.Strings;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * An adapter that implements a valid EAP 8 {@link HelmChartRelease} by exposing an internal instance of
+ * {@link HelmEap81Release} to store an EAP 8 Helm Charts release data and to represent a values file which can be serialized
+ * as an output for {@link HelmChartRelease#toValuesFile()}
+ *
+ * This adapter is compliant with the contract which is required by the
+ * {@link org.jboss.intersmash.provision.helm.HelmChartOpenShiftProvisioner} logic, i.e. to
+ * implement {@link HelmChartRelease}, and allows for us to leverage a generated
+ * {@link Eap81HelmChartReleaseAdapter#adaptee}, i.e. in terms of UX, provide native release YAML definitions.
+ */
+@Slf4j
+public class Eap81HelmChartReleaseAdapter extends HelmChartReleaseAdapter<HelmEap81Release>
+		implements HelmChartRelease, WildflyHelmChartRelease {
+
+	public Eap81HelmChartReleaseAdapter(@NonNull HelmEap81Release release) {
+		super(release, new ArrayList<>());
+		initDefaultReplicas(release);
+	}
+
+	public Eap81HelmChartReleaseAdapter(@NonNull HelmEap81Release release, List<Path> additionalValuesFiles) {
+		super(release, additionalValuesFiles);
+		initDefaultReplicas(release);
+	}
+
+	@Override
+	public Map<String, String> getDeploymentEnvironmentVariables() {
+		return adaptee.getDeploy() == null ? Map.of()
+				: adaptee.getDeploy().getEnv() == null || adaptee.getDeploy().getEnv().isEmpty() ? Map.of()
+						: adaptee.getDeploy().getEnv().stream()
+								.collect(Collectors.toMap(e -> e.getName(), e -> e.getValue()));
+	}
+
+	@Override
+	public void setDeploymentEnvironmentVariables(Map<String, String> deploymentEnvironmentVariables) {
+		if (adaptee.getDeploy() == null) {
+			adaptee.setDeploy(new Deploy());
+		}
+		if ((adaptee.getDeploy().getEnv() == null) || adaptee.getDeploy().getEnv().isEmpty()) {
+			adaptee.getDeploy().setEnv(new ArrayList<>());
+		} else {
+			adaptee.getDeploy().getEnv().clear();
+		}
+		// let's check just here, so there's a way to clear, i.e. by passing an empty or null list
+		if ((deploymentEnvironmentVariables != null) && !deploymentEnvironmentVariables.isEmpty()) {
+			deploymentEnvironmentVariables.entrySet().stream()
+					.forEach(e -> adaptee.getDeploy().getEnv().add(new Env__1().withName(e.getKey()).withValue(e.getValue())));
+		}
+	}
+
+	public WildflyHelmChartRelease withDeploymentEnvironmentVariables(
```

Comment:
Can you please add javadoc at least to non-overrided & non-constructor methods here?